File: app/src/main/java/io/connorwyatt/flashcards/data/entities/Flashcard.java
package io.connorwyatt.flashcards.data.entities;

import java.util.ArrayList;
import java.util.List;

public class Flashcard extends BaseEntity {
    private String title;
    private String text;
    private List<Category> categories = new ArrayList<>();

    public Flashcard() {
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public List<Category> getCategories() {
        return categories;
    }

    public void setCategories(List<Category> categories) {
        this.categories = categories;
    }

    public String getCategoriesString() {
        String categoriesString = "";

        for (Category category : categories) {
            if (categoriesString.length() != 0) {
                categoriesString = categoriesString.concat(", ");
            }

            categoriesString = categoriesString.concat(category.getName());
        }

        return categoriesString;
    }
}

File: app/src/main/java/io/connorwyatt/flashcards/data/entities/Flashcard.kt
package io.connorwyatt.flashcards.data.entities

import com.google.firebase.database.DataSnapshot

class Flashcard(data: DataSnapshot?) : BaseEntity(data)
{
    var title: String? = null
    var text: String? = null

    init
    {
        val values = data?.value as? Map<*, *>

        title = values?.get(PropertyKeys.title) as String?
        text = values?.get(PropertyKeys.text) as String?
    }

    override fun serialise(): MutableMap<String, Any?>
    {
        val serialisedEntity = super.serialise()

        serialisedEntity.put(PropertyKeys.title, title)
        serialisedEntity.put(PropertyKeys.text, text)

        return serialisedEntity
    }

    companion object
    {
        object PropertyKeys
        {
            val title = "title"
            val text = "text"
        }
    }
}
